### PR TITLE
Cleanup makefiles

### PR DESCRIPTION
**Pull request by Kubuxu**
This can wait for after release but it shouldn't affect the release in any form either.

We might want to limit number of threads on the sharness or make output sequential of sharness because now they run in parallel.

### CODE REVIEW

File: test/Makefile [hunk 1]
```diff
@@ -15,12 +15,13 @@ GOFLAGS =
...
 global-deps:
-	make -C .. deps
+	$(MAKE) -C .. deps
 
 deps: global-deps bins
 
 clean:
-	rm $(BINS)
+	$(MAKE) -C sharness clean
+	-rm -rf $(BINS)
```
Comment (lgierth) [inline test/Makefile]: what's `-rm` doing?

**Comment by lgierth** [comment]
LGTM apart from one question ^

File: test/Makefile (hunk 1 above)
Comment (Kubuxu) [inline test/Makefile]: `-` prefix makes `make` ignore return code of `rm` command.

Comment (lgierth) [inline test/Makefile]: Okay then I think it's not needed -- `-f` makes it ignore non-existing files/dirs anyhow

**Comment by thomas-gardner** [comment]
You can squash these:

```
build install nofuse: deps
        $(MAKE) -C cmd/ipfs $@

clean uninstall:
        $(MAKE) -C cmd/ipfs $@
```

And if we could drop "deps" as a prerequisite here (it is serious overkill), all 5 options could be squashed into two lines.

**Comment by Kubuxu** [comment]
@thomas-gardner that is awesome trick, Thanks.

File: test/Makefile (hunk 1 above)
Comment (Kubuxu) [inline test/Makefile]: Yup, I should clear it up everywhere.

**Comment by whyrusleeping** [comment]
@Kubuxu wanna go ahead and rebase this on master?

**Comment by Kubuxu** [comment]
done

**Pull request merged**